cls/SourceControl/Git/Utils.cls

```diff
@@ -3043,4 +3042,77 @@ ClassMethod InDefaultBranchBasicMode() As %Boolean
     quit 0
 }
 
+/// Runs the commit on the specified files (unstaging and restaging currently staged files)
+ClassMethod RunGitCommandReStage(Output outStream, Output errStream, command As %String, ByRef fileList As %Library.DynamicArray, args...) As %Integer
+{
+    set sc = $$$OK
+    LOCK +^gitCommit
```

Comment:
Missed on earlier review: we're not really following locking best practices here. See: https://community.intersystems.com/post/robust-error-handling-and-cleanup-objectscript